common/utils/ti_fec.cpp

```diff
@@ -0,0 +1,28 @@
+#include "ti_fec.h"
```

Comment:
Can you rearrange/rename some of the `ti_fec*` files? For the ones that are classes, it should probably be one header file for the class and then a source file. It looks like there's also some general utilities, which can be its own header/source. The current naming scheme doesn't make it immediately clear what is located where.